learningresources/tests/test_api.py

```diff
@@ -190,5 +184,105 @@ def test_create_static_asset(self):
             test_file.close()
             test_file.name = 'new_name'
             with self.assertRaises(ValueError):
-                asset = create_static_asset(self.course.id, test_file)
+                asset = api.create_static_asset(self.course.id, test_file)
         self.assertEqual(file_contents, asset.asset.read())
+
+
+class TestDescriptionPath(LoreTestCase):
+    """
+    Tests the Python API for the description paths
+    """
+    def test_update_description_path(self):
+        """Tests for update_description_path"""
+        # after created a resource without parent has the description path
+        # equal to the title
+        self.assertIsNone(self.resource.parent)
+        self.assertEqual(
+            self.resource.title,
+            self.resource.description_path
+        )
+        # changing the title does not update the descriptio path automatically
+        self.resource.title = '123 xyz'
+        self.resource.save()
+        self.assertNotEqual(
+            self.resource.title,
+            self.resource.description_path
+        )
+        # update the description path
+        api.update_description_path(self.resource)
+        self.assertEqual(
+            self.resource.title,
+            self.resource.description_path
+        )
+        # create a child resource
+        child_res = self.create_resource(
+            parent=self.resource
+        )
+        # the description path is the combination of the child resource title
+        # and the parent description path
+        self.assertEqual(
+            child_res.description_path,
+            api.join_description_paths(
```

Comment:
In any of the invocations of `join_description_paths` are we passing in an empty string? If not we should assert that the empty string is skipped